metrics.go

```diff
@@ -0,0 +1,217 @@
+package cke
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	v3 "github.com/coreos/etcd/clientv3"
+	"github.com/cybozu-go/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type logger struct{}
+
+func (l logger) Println(v ...interface{}) {
+	log.Error(fmt.Sprint(v...), nil)
+}
+
+const (
+	namespace     = "cke"
+	scrapeTimeout = time.Second * 10
+)
+
+// Metric represents collector and availability of metric.
+type Metric struct {
+	collectors  []prometheus.Collector
+	isAvailable func(context.Context, *Storage) (bool, error)
+}
+
+// Collector is a metrics collector for CKE.
+type Collector struct {
+	metrics map[string]Metric
+	storage *Storage
+}
+
+// NewCollector returns a new Collector.
+func NewCollector(client *v3.Client) *Collector {
+	return &Collector{
+		metrics: map[string]Metric{
+			"operation_phase": {
+				collectors:  []prometheus.Collector{operationPhase, operationPhaseTimestampSeconds},
+				isAvailable: alwaysAvailable,
```

Comment:
Really?  I think this should only be available when the instance is the leader.

Reply:
That's right.  I'm updating the code and adding the following note to the doc.

> All metrics but `leader` are available only when the server is the leader of CKE.